### DISCUSSION TITLE
JLCPCB assembly question

### DISCUSSION THREAD

**Discussion by brentward** [picocomputer/community#19] [Q&A]
Forgive me if I'm asking asking obvious questions or making some kind of stupid mistake. I've done lots of soldering and breadboard electronics/computers stuff but I've never ordered a custom PCB.

I'm trying to price out getting a board made and soldering it myself vs having them assemble it. When I try to get a quote from JLCPCB it requires a BOM file, which I have, and a CPL file, which I don't have and I believe is a pick and place location file. According to JLCPCB's help section the pick and place location file is required for surface mount and through hole placement. Am I doing something wrong with the order or am I missing this file?

I've also tried getting quotes from PCBWay and for some reason it seems to be stuck in review. It says it should take 10 minutes and it has been a couple of hours. Again, I'm not sure if I'm doing something wrong or if it sometimes just takes longer to review.

Edit: It seems like the centroid file is the pick and place location data file, which the documentation says is not needed. Does anyone know how I get my JLCPCB order to not require the CPL file for processing?

**Comment by rumbledethumps** [comment]
PCBWay can tell you why their time estimate is off if you ask them. JLCPCB can help you with whatever mysterious forces are requiring a centroid file for a project without surface mount parts, but you have to ask them.

**Reply by brentward** [reply]
I gave up on the JLCPCB thing because I wanted to solder it myself anyway and I was just doing to to see what the cost difference was to have them get all the parts for me. I ended up just ordering the board without assembly which was working fine in both JLCPCB and PCBWay. I should have everything soon. I'm looking forward to getting it running.

**Reply by ronaldlens** [reply]
I just received my soldered board from PCBWay. I explained them in the communications that there was no centroid file due to no smd parts. This was fine. They even sent pictures before the soldered the parts. Not cheap (USD125 for the whole thing incl. shipping) but I was lazy:)
![IMG_7475](https://github.com/picocomputer/discussions/assets/2178191/a8474ec9-87a3-4a99-bf73-19cf63a05b62)
